specification.md

```diff
@@ -2508,10 +2508,25 @@ to the trigger/produced event.
 
 #### SubFlowRef Definition
 
+`subFlowRef` definition can have two types, either `string` or `object`.
+If `string` type, it defines the unique id of the sub-workflow to be invoked. 
+This short-hand definition can be used if sub-workflow lookup is done only by its `id`
+property and not it's `version` property and if the default value of `waitForCompletion` is assumed.
+
+```json
+{
+  "subFlowRef": "uniqueSubFlowId"
```

Comment:
Should we consider a mechanism for including the version here in the short-hand?  IMHO, best-practices for production workflows should be to always specify the version (for repeatability), so that would preclude production workflows from using this short-hand if so.

How about allowing the format `<workflowId>:<version>`?  This would obviously require that the workflow id now allow whatever delimiter character is chosen, to avoid confusion.

Reply:
yeah, since we dont enforce any specific version strategy doing this can be pretty tricky to do

Reply:
+1 on having concrete version as recommendation as it makes it explicit and you can directly know what is behind. 

I wouldn’t go for id:version syntax as this is easy to miss. Having it as separate attribute is in my opinion the way to go. 

And lastly about version on top level workflow I’d go with it the same as subflows to be consistent.

Reply:
@mswiderski 
> +1 on having concrete version as recommendation as it makes it explicit and you can directly know what is behind.

I'm not sure if I'm misinterpreting what you're saying, but it seems like you're saying that the version property should be required?  If so, I agree.  I think that's preferable to having it be optional if the default value is expected to be 'latest'.

> And lastly about version on top level workflow I’d go with it the same as subflows to be consistent.

I'm not following what you meant here, could you expand?

Reply:
I prefer the spec does not enforce versions to be used as not always there is a need to version your workflows and there should be really good reason to change versions as in case of stateful workflows this can become quite a headache. So having it recommended to be explicitly given if you want to use versions and defaults to `latest` (that in fact will be resolved by runtimes) is the middle ground in my opinion. The same should apply to both top level workflows and sub flows.

Reply:
@mswiderski 

> defaults to `latest` (that in fact will be resolved by runtimes)

It sounds like you're saying that you're ok with the runtimes determining their definition of 'latest'.  In that case, I'd be fine with that if we could choose a term other than 'latest', given that 'latest' implies a particular strategy for resolving the version.

Perhaps 'live' or 'default'?

Reply:
in general we could use something else but I think latest is sort of well established especially for cloud targeting deployments e.g. containers. But I don't have strong preference to keep latest or something else. Maybe except the default which we will have same issue as it is reserver word in many languages.

Reply:
I don't know for sure, but 'default' in this context is a value (as opposed to an attribute name) so I don't think there would be a problem with reserved-word conflicts as this would just be the string "default", right?

Reply:
yes, you're right, with it as a value is less of an issue.